### PR TITLE
[IOTDB-3000] ConfigNode supports MTreeAboveSG

### PR DESCRIPTION
**Pull request by CRZbulabula**
To manage StorageGroupSchema more formally, ConfigNode supports the use of MTreeAboveSG. ConfigNode also provides DataNode with more StorageGroup-related RPC interfaces.

- You can see [ConfigNode for DataNode RPC interfaces](https://y8dp9fjm8f.feishu.cn/docs/doccnCp13O7QDU7O6GMvagUwEDc) for RPC changes.

### CODE REVIEW

File: thrift-confignode/src/main/thrift/confignode.thrift [hunk 1]
```diff
@@ -62,11 +62,31 @@ struct TSetTTLReq {
...
+  1: required string storageGroup
+  2: required i32 dataReplicationFactor
+}
+
 struct TSetTimePartitionIntervalReq {
   1: required string storageGroup
   2: required i64 timePartitionInterval
 }
 
+struct TStorageGroupReq {
```
Comment (qiaojialin) [inline thrift-confignode/src/main/thrift/confignode.thrift:81]: TPathPatternReq

File: confignode/src/main/java/org/apache/iotdb/confignode/persistence/StorageGroupInfo.java [hunk 2]
```diff
@@ -35,30 +49,137 @@
 
 public class StorageGroupInfo {
```
Comment (qiaojialin) [inline confignode/src/main/java/org/apache/iotdb/confignode/persistence/StorageGroupInfo.java:50]:
```suggestion
public class ClusterSchemaInfo {
```

File: thrift-confignode/src/main/thrift/confignode.thrift (hunk 1 above)
Comment (CRZbulabula) [reply thrift-confignode/src/main/thrift/confignode.thrift]: Fixed.

File: confignode/src/main/java/org/apache/iotdb/confignode/persistence/StorageGroupInfo.java (hunk 2 above)
Comment (CRZbulabula) [reply confignode/src/main/java/org/apache/iotdb/confignode/persistence/StorageGroupInfo.java]: Fixed.

**Review by SpriCoder — APPROVED** [review]
LGTM~

**Review by qiaojialin — APPROVED** [review]

File: confignode/src/main/java/org/apache/iotdb/confignode/persistence/ClusterSchemaInfo.java [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+  }
+
+  /**
+   * Persistence new StorageGroupSchema
+   *
+   * @param req SetStorageGroupReq
+   * @return SUCCESS_STATUS
+   */
+  public TSStatus setStorageGroup(SetStorageGroupReq req) {
+    TSStatus result;
```
Comment (wangchao316) [inline confignode/src/main/java/org/apache/iotdb/confignode/persistence/ClusterSchemaInfo.java:76]: TSStatus result = new TSStatus();

if success,  "result .setCode(TSStatusCode.SUCCESS_STATUS.getStatusCode());"

File: confignode/src/main/java/org/apache/iotdb/confignode/persistence/ClusterSchemaInfo.java [hunk 4]
```diff
@@ -0,0 +1,313 @@
...
+      PartialPath path = new PartialPath(req.getStorageGroup());
+      if (mTree.isStorageGroupAlreadySet(path)) {
+        mTree.getStorageGroupNodeByPath(path).getStorageGroupSchema().setTTL(req.getTTL());
+        result = new TSStatus(TSStatusCode.SUCCESS_STATUS.getStatusCode());
+      } else {
+        result = new TSStatus(TSStatusCode.STORAGE_GROUP_NOT_EXIST.getStatusCode());
+      }
+    } catch (MetadataException e) {
+      LOGGER.error("Error StorageGroup name", e);
+      return new TSStatus(TSStatusCode.STORAGE_GROUP_NOT_EXIST.getStatusCode())
```
Comment (wangchao316) [inline confignode/src/main/java/org/apache/iotdb/confignode/persistence/ClusterSchemaInfo.java:109]: the same as above

**Review by wangchao316 — CHANGES_REQUESTED** [review]

File: confignode/src/main/java/org/apache/iotdb/confignode/persistence/ClusterSchemaInfo.java (hunk 3 above)
Comment (CRZbulabula) [reply confignode/src/main/java/org/apache/iotdb/confignode/persistence/ClusterSchemaInfo.java]: Fixed.

File: confignode/src/main/java/org/apache/iotdb/confignode/persistence/ClusterSchemaInfo.java (hunk 4 above)
Comment (CRZbulabula) [reply confignode/src/main/java/org/apache/iotdb/confignode/persistence/ClusterSchemaInfo.java]: Fixed.

**Review by wangchao316 — APPROVED** [review]
LGTM

**Pull request merged**